lib/search/search_page.dart

```diff
@@ -43,6 +56,13 @@ class SearchPage extends ConsumerWidget {
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
+    final searchButtonEnabled = ref.watch(searchButtonStateProvider);
+
+    searchCtrl.addListener(() {
+      ref.read(searchButtonStateProvider.notifier).setEnabled(
+            searchCtrl.text.length > 0,
+          );
+    });
```

Comment:
Let's move this to onChanged because it is called on the same change action:


   onChanged: (v) {
                                          _formKey.currentState?.validate();
                                        },